android/app/src/main/java/com/reactnativenavigation/views/slidingOverlay/SlidingOverlaysQueue.java

```diff
@@ -60,23 +59,27 @@ else if (currentOverlay.isVisible()) {
 
     @Override
     public void onSlidingOverlayShown() {
-        int autoDismissDuration = pendingHide || queue.size() > 1
-                ? SHORT_SUSTAIN_DURATION
-                : SUSTAIN_DURATION;
-        pendingHide = false;
+        Integer autoDismissTimerSec = queue.peek().getAutoDismissTimerSec();
 
-        autoDismissTimer = new Timer();
-        autoDismissTimer.schedule(new TimerTask() {
-            @Override
-            public void run() {
-                NavigationApplication.instance.runOnMainThread(new Runnable() {
-                    @Override
-                    public void run() {
-                        queue.peek().hide();
-                    }
-                });
-            }
-        }, autoDismissDuration);
+        if (autoDismissTimerSec != null || pendingHide || queue.size() > 1) {
+            int autoDismissDuration = autoDismissTimerSec != null
+                    ? autoDismissTimerSec * 1000
+                    : SHORT_SUSTAIN_DURATION;
+            pendingHide = false;
```

Comment:
But, I'm not sure interrupting the show animation is always feasible/aesthetically pleasing (depending on the type of animation. e.g. Flip)